### PR TITLE
Lots of small documentation updates

### LINKED ISSUE #42: Include saving model in docs/tutorial
Can an example of serializing models and datasets to disk be included in the tutorials and/or docs? Additional help on using ONNX would be helpful too.

### PR DESCRIPTION
**Pull request by Craigacp**
### Description
Many small javadoc fixes, general docs fixes, tutorial updates, new tutorials on columnar data loading and external model imports.

In addition to docs fixes there are three bug fixes: `MultiLabelEvaluationImpl.toString()` had the same accuracy integer division bug as `LabelEvaluation.toFormattedString()`, `Event.equals` and `Event.hashCode` looked at the score when they are documented not to (though this is less relevant for those classes as the domains are specialised because there are only three possible values), and `DummyRegressionTrainer` was using an old provenance class not `TrainerProvenanceImpl` like it should be.

Finally it patches the pom file parent reference to fix an issue with project discovery.

### Motivation
Fixes for the docs found by our users after release.

Fixes #42, #50, #53, #60, #77.

### CODE REVIEW

File: AnomalyDetection/Core/src/main/java/org/tribuo/anomaly/AnomalyInfo.java [hunk 1]
```diff
@@ -36,12 +36,30 @@
...
+     */
     protected long expectedCount = 0;
+
+    /**
+     * The number of anomalous events observed.
+     */
     protected long anomalyCount = 0;
+
+    /**
+     * The number of unknown events observed (i.e. those without labels).
```
Comment (eelstretching) [inline AnomalyDetection/Core/src/main/java/org/tribuo/anomaly/AnomalyInfo.java:50]: Comma after i.e.

File: Classification/Core/src/main/java/org/tribuo/classification/baseline/DummyClassifierTrainer.java [hunk 2]
```diff
@@ -39,7 +39,27 @@
  */
 public final class DummyClassifierTrainer implements Trainer<Label> {
```
Comment (eelstretching) [inline Classification/Core/src/main/java/org/tribuo/classification/baseline/DummyClassifierTrainer.java:40]: Not super happy with Dummy? Maybe BaselineClassifierTrainer would be better? Not specifically introduced by this PR, but maybe we should fix it while we're here?

File: Common/XGBoost/src/main/java/org/tribuo/common/xgboost/XGBoostTrainer.java [hunk 3]
```diff
@@ -63,7 +63,8 @@
  * </pre>
  * N.B.: This uses a native C implementation of xgboost that links to various C libraries, including libgomp
  * and glibc. If you're running on Alpine, which does not natively use glibc, you'll need to install glibc
- * into the container.
+ * into the container. On Windows this binary is not available in the Maven Central release, you'll need
+ * to compile it from source.
```
Comment (eelstretching) [inline Common/XGBoost/src/main/java/org/tribuo/common/xgboost/XGBoostTrainer.java:67]: Do we have any extant advice on compiling XGBoost? Could we link to some tribuo.org page on how to do it? I guess we don't want to support the myriad Windows installs out there ("I couldn't get it to work on Windows ME!")

File: Core/src/main/java/org/tribuo/Example.java [hunk 4]
```diff
@@ -47,8 +47,14 @@
...
 
+    /**
+     * The default initial size of the metadata map.
+     */
     protected static final int DEFAULT_METADATA_SIZE = 2;
 
+    /**
+     * The default weight.
+     */
     public static final float DEFAULT_WEIGHT = 1.0f;
```
Comment (eelstretching) [inline Core/src/main/java/org/tribuo/Example.java:58]: What's the coding standard for numbers like 1 and 0. If I recall correctly, just using 1 and 0 will get you the 1 or 0 of the right type. Having it be explicit is probably better?

File: Data/src/main/java/org/tribuo/data/csv/package-info.java [hunk 5]
```diff
@@ -17,5 +17,12 @@
 /**
  * Provides classes which can load columnar data (using a {@link org.tribuo.data.columnar.RowProcessor})
  * from a CSV (or other character delimited format) file.
+ * <p>
+ * {@link org.tribuo.data.csv.CSVDataSource} is the main way of loading CSV format data into Tribuo.
+ * It provides full control over featurisation, output processing and metadata extraction.
+ * {@link org.tribuo.data.csv.CSVLoader} is for simple numerical CSV files where all the
+ * non-response columns should be treated as features. {@link org.tribuo.data.csv.CSVSaver} writes
+ * out a * Tribuo {@link org.tribuo.Dataset} in CSV format suitable for loading via
```
Comment (eelstretching) [inline Data/src/main/java/org/tribuo/data/csv/package-info.java:25]: What's that stray asterisk doing in there?

**Review by eelstretching — CHANGES_REQUESTED** [review]
Address comments in the PR.

File: Classification/Core/src/main/java/org/tribuo/classification/baseline/DummyClassifierTrainer.java (hunk 2 above)
Comment (Craigacp) [reply Classification/Core/src/main/java/org/tribuo/classification/baseline/DummyClassifierTrainer.java:40]: It mirrors scikit-learn's names for these kind of classifiers/regressors. We can change it in the next major version bump if you want.

File: Common/XGBoost/src/main/java/org/tribuo/common/xgboost/XGBoostTrainer.java (hunk 3 above)
Comment (Craigacp) [reply Common/XGBoost/src/main/java/org/tribuo/common/xgboost/XGBoostTrainer.java:67]: Basically you need to install the Windows SDK to get the msbuild system and then run mvn package in the jvm-packages directory. I can talk to the XGBoost people about it and maybe firm up some instructions.

File: Core/src/main/java/org/tribuo/Example.java (hunk 4 above)
Comment (Craigacp) [reply Core/src/main/java/org/tribuo/Example.java:58]: I tend to put the specifier if it's a long, or if it's a float, and will put `d` on things if they are integers assigned to longs, but I usually prefer to put the decimal place in that case.

File: Data/src/main/java/org/tribuo/data/csv/package-info.java (hunk 5 above)
Comment (Craigacp) [reply Data/src/main/java/org/tribuo/data/csv/package-info.java:25]: Oops, merged the lines in vim. Fixed.

File: AnomalyDetection/Core/src/main/java/org/tribuo/anomaly/AnomalyInfo.java (hunk 1 above)
Comment (Craigacp) [reply AnomalyDetection/Core/src/main/java/org/tribuo/anomaly/AnomalyInfo.java:50]: Fixed.

**Comment by eelstretching** [comment]
Looks good to me.

**Review by eelstretching — COMMENTED** [review]
LGTM

**Review by eelstretching — APPROVED** [review]
LGTM
